### PR TITLE
Faster Makefile dist build, less crud

### PR DESCRIPTION
**Pull request by PVince81**
## Description
- When building with "make dist", use a shortcut and only install bower as we're not going to run any unit tests or so. (useful for the release scripts)
- Removed more unneeded release files from composer and co.
## Motivation and Context

Making release scripts run faster without installing extra unused npm deps.
Less unneeded files in the build.
## How Has This Been Tested?

Run `make dist-dir` and see that only bower is installed by npm and it runs fine.
Then check the build folder "build/dist/owncloud" and search for test files like '*Spec.js' or even exe files (symfony console...).
Before this fix: still a lot of such files.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

@DeepDiver1975 @butonic @danimo

### CODE REVIEW

**Comment by PVince81** [comment]
Ready for review @DeepDiver1975 @butonic @jvillafanez

File: Makefile [hunk 1]
```diff
@@ -80,6 +80,11 @@ clean-composer-deps:
 $(nodejs_deps): build/package.json
 	$(NPM) install --prefix $(NODE_PREFIX)
 
+# in some cases installing bower alone could be enough
+$(BOWER):
+	$(NPM) install --prefix $(NODE_PREFIX) bower
```
Comment (jvillafanez) [inline Makefile]: Maybe quote the `$(NPM)` to make sure spaces are handle correctly? I think that's the only path we don't control.

Comment (PVince81) [inline Makefile]: In theory we don't control any paths. Make allows callers to override any of the variables above.
For example `make BOWER=/my/path/to/bower` would override the "BOWER" variable

Comment (jvillafanez) [inline Makefile]: I'm worried about weird paths. If someone has the binaries in "/home/weird user/custom binaries/bin", would the make command handle those paths properly?

Comment (PVince81) [inline Makefile]: Let's do some testing and address these separately for all variables...

Comment (PVince81) [inline Makefile]: @jvillafanez raised https://github.com/owncloud/core/issues/26035

Can we merge this ?

**Comment by jvillafanez** [comment]
👍

**Pull request merged**